### DISCUSSION TITLE
Create Issued Document - mantenere il prodotto in stock su FIC

### DISCUSSION THREAD

**Discussion by Picaland** [fattureincloud/api#409] [Q&A] [answered]
Buongiorno a tutti,
Volevo sapere se il parametro "stock" su items_list serve a mantenere lo stato in stock sul prodotto anche se va sottoscorta.
un cliente mi lamenta che se il prodotto va sottoscorta viene staccata la gestione magazzino del prodotto.
come possiamo evitare il problema?

<img width="810" alt="Screenshot 2024-02-08 alle 11 28 14" src="https://github.com/fattureincloud/api/assets/5677228/60429ab9-278b-498b-9035-4052e9d14e80">
<img width="295" alt="Screenshot 2024-02-08 alle 11 29 36" src="https://github.com/fattureincloud/api/assets/5677228/6fb7a362-3108-4adc-8b5d-7ecf37aed258">

grazie!

**Comment by emajo** [comment]
Ciao @Picaland, ho fatto un test e non risulta questo comportamento, il magazzino può anche andare in negativo e non i prodotti non vengono disattivati dal magazzino, non è un comportamento magari implementato a livello di integrazione?

Grazie,
Emanuele.

**Reply by Picaland** [reply]
Ciao @emajo, ritorno nuovamente da te.
hai detto "non è un comportamento magari implementato a livello di integrazione?" mi sapresti dire cosa potrebbe causare questo comportamento ? 
considerando che l'integrazione in relazione ai prodotti fa questo

1. scarica qty in fase di creazione fattura 
2. modifica qty su FIC quando su WooCommerce viene aggiornato il prodotto
3. modifica qty su FIC quando su WooCommerce viene fatto un rimborso.

**Reply by emajo** [reply]
Non ho idea di cosa possa causare questo comportamento se le operazioni effettuate sono quelle elencate, ti direi di controllare i log dell'integrazione oppure scrivere al nostro supporto clienti per controllare le operazioni effettuate dall'utilizzatore dell'integrazione.

**Comment by Picaland** [comment]
Ciao @emajo, intanto grazie per il tempo che ci hai dedicato, ma a questo punto sono ancora più confuso :) 
se non erro la qty del prodotto viene scaricata quando viene creata la fattura, corretto ?

l'oggetto items_list che mando è composto in questo modo:

      // Items data
      $body['data']['items_list'][] = array(
          'vat'                     => $vatType,
          'not_taxable'             => false,
          'apply_withholding_taxes' => false,
          'discount'                => $discountPercentage,
          'discount_highlight'      => false,
          'in_ddt'                  => false,
          'product_id'              => (int)$prodID ?: 000,
          'category'                => $category ?: "",
          'code'                    => $prodCode,
          'name'                    => $prodName,
          'description'             => apply_filters('wfc-invoice-items_list_description', '', $prodShotDesc, $prodDesc),
          'net_price'               => floatval(end($item->PrezzoUnitario)),
          'gross_price'             => floatval(end($item->PrezzoTotale)),
          'qty'                     => apply_filters('wfc-invoice-items_list_qty', (int)end($item->Quantita), $prodID),
      );

Quindi dalla tua risposta deduco che quando creo la fattura anche se lo stock va in negativo non viene staccato ?
Se è cosi allora devo chiedere al cliente cosa fa in realtà per avere questo risultato.

**Reply by emajo** [reply]
Esatto, i prodotti vengono scaricati in creazione del documento, ma se appunto abbiamo un prodotto con qty = 5 e ne scarichiamo 10 questo prodotto avrà a stock -10 di qty, non andiamo mai a staccare la gestione dello stock lato nostro.

**Comment by valmoz** [comment]
Buongiorno @Picaland posso chiudere la discussione?

Grazie
Mauro

**Reply by Picaland** [reply, marked as answer]
Buongiorno @valmoz 
scusami per il mancato aggiornamento ma forse abbiamo capito la problematica.
la nostra integrazione per la gestione dello stock prevede che a determinate condizioni viene fatta una richiesta che modifica il prodotto tramite new ModifyProductRequest($args).

partendo dall'esempio che trovo https://developers.fattureincloud.it/api-reference/#put-/c/-company_id-/products/-product_id-
{
  "data": {
    "name": "Tavolo di marmo",
    "code": "TAVOLO003",
    "net_price": 240,
    "net_cost": 0,
    "measure": "",
    "description": "Tavolo in marmo pregiato",
    "category": "arredamento",
    "in_stock": true,
    "default_vat": {
      "id": 3,
      "value": 22,
      "description": "Non imponibile art. 123",
      "notes": "IVA non imponibile ai sensi dell'articolo 123, comma 2",
      "is_disabled": false
    }
  }
}

se mando "in_stock": false, si verifica la problematica. 
mi confermi ?

**Reply by emajo** [reply]
Ciao @Picaland si, effettuando questa modifica ha senso si verifichino gli effetti rilevati.

**Reply by Picaland** [reply]
@emajo 
Ah ecco, ok grazie!
potete chiudere la discussione

Grazie ancora!
